src/rust/src/backend/rsa.rs

```diff
@@ -290,8 +290,16 @@ impl RsaPrivateKey {
         padding: &pyo3::Bound<'p, pyo3::PyAny>,
         algorithm: &pyo3::Bound<'p, pyo3::PyAny>,
     ) -> CryptographyResult<pyo3::Bound<'p, pyo3::types::PyAny>> {
-        let (data, algorithm) =
-            utils::calculate_digest_and_algorithm(py, data.as_bytes(), algorithm)?;
+        let (data, algorithm) = {
```

Comment:
What happens if I pass NoDigestInfo and also PSS instead of PKCS1v15 for padding? Do we properly handle that case (it should be a ValueError that NoDigestInfo is only allowed for PKCS1v15)